rethinkdb/datadog_checks/rethinkdb/document_db/query.py

```diff
@@ -0,0 +1,120 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import logging
+from typing import Any, Callable, Iterator, List, Mapping, Sequence, Tuple
+
+from datadog_checks.base import AgentCheck
+
+from .types import Enumeration, Group, Metric, MetricSpec, Modifier
+from .utils import dotted_join, lookup_dotted, to_timestamp
+
+logger = logging.getLogger(__name__)
+
+
+class DocumentQuery(object):
```

Comment:
Note to reviewers: this component roughly follows the same idea than `QueryManager`, but for document databases.

Eventually this might end up as a util in `datadog_checks_base` (or merged into `QueryManager` (?)), so I isolated it into its own sub-package here.